bindings/Python/CMakeLists.txt

```diff
@@ -13,16 +13,29 @@ if(ADIOS2_HAVE_MPI)
   target_link_libraries(adios2py PRIVATE PythonModule::mpi4py)
 endif()
 
-set(CMAKE_INSTALL_PYTHONDIR
-  "${CMAKE_INSTALL_LIBDIR}/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages"
+if(WIN32)
+  set(CMAKE_INSTALL_PYTHONDIR_DEFAULT
+    "${CMAKE_INSTALL_LIBDIR}\\site-packages"
+  )
+else()
+  set(CMAKE_INSTALL_PYTHONDIR_DEFAULT
+    "${CMAKE_INSTALL_LIBDIR}/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages"
+  )
+endif()
+set(CMAKE_INSTALL_PYTHONDIR "${CMAKE_INSTALL_PYTHONDIR_DEFAULT}"
+  CACHE PATH "Install directory for python modules"
 )
-set(CMAKE_INSTALL_PYTHONDIR "${CMAKE_INSTALL_PYTHONDIR}"
-  CACHE INTERNAL "" FORCE
+set(CMAKE_PYTHON_OUTPUT_DIRECTORY
+  "${ADIOS2_BINARY_DIR}/${CMAKE_INSTALL_PYTHONDIR}"
+  CACHE PATH "Build directory for python modules"
```

Comment:
Have you tried this with multi-config generators like Visual studio.  This often doesn't quite work since you need to be able to hole release and debug builds simultaneously in the same build tree.